packages/components/src/popover/index.tsx

```diff
@@ -488,8 +488,15 @@ const UnforwardedPopover = (
 					? undefined
 					: {
 							position: strategy,
-							left: Number.isNaN( x ) ? 0 : x ?? undefined,
-							top: Number.isNaN( y ) ? 0 : y ?? undefined,
+							top: 0,
+							left: 0,
+							// `x` and `y` are framer-motion specific props and are shorthands
+							// for `translateX` and `translateY`. Currently it is not possible
+							// to use `translateX` and `translateY` because those values would
+							// be overridden by the return value of the
+							// `placementToMotionAnimationProps` function in `AnimatedWrapper`
+							x: Math.round( x ?? 0 ) || undefined,
```

Comment:
Previously, while animating we were accepting decimal values for `x` and `y`, and now we're rounding them. This is likely the reason for a bunch of e2e tests failing.

Reply:
@tyxla Yes it was suggested by @ciampo here : https://github.com/WordPress/gutenberg/pull/45545#discussion_r1035127580

Reply:
@tyxla The tests indeed fail because of this but I have no idea how to change the expected values

Reply:
`npm run test:unit:update` should update the snapshots — you should then commit those changes

Reply:
@ciampo I ran it but there nothing to commit 🤔

Reply:
Do you mind rebasing and fixing conflicts? I'm currently unable to see that tests are failing, which makes it harder to make an informed suggestion 😅

Reply:
@tyxla I can see unit tests erroring because of the `act` warning — do you mind helping out here? (I'm going to be AFK in the following weeks)

Reply:
Sure thing! Happy to aid with that next week.

Reply:
It looks like the change here results in `0` values not being applied to update popover positions for popovers that are intended to be flush with the edge of the screen. I've opened up a potential fix over in #51320.